fix: revert slash validation and change validation message

danez opened the pull request:
Adjust the validation message as proposed in #339

klavavej commented on node/validation/manifest/schema.ts:
```diff
@@ -17,8 +17,7 @@ const routesSchema = {
       format: 'regexPattern',
-      errorMessage:
-        'pattern needs to be a regex that starts with ^ followed by / and ends with $ without any additional slashes before and afterwards',
+      errorMessage: 'pattern must be a regex that starts with ^/ and ends with $ for example ^/blog/[d]{4}$',
```
+1 to @eduardoboucas 's great idea about introducing some separation here. But, it looks like there might be an extraneous `)` in the suggestion he added (especially when compared to his similar suggestion below) so here's a new one without that

```suggestion
      errorMessage: 'pattern must be a regex that starts with ^/ and ends with $ (e.g. ^/blog/[d]{4}$)',
```

klavavej commented on node/validation/manifest/schema.ts:
```diff
@@ -35,7 +34,7 @@ const functionConfigSchema = {
         format: 'regexPattern',
         errorMessage:
-          'excluded_patterns needs to be an array of regex that starts with ^ followed by / and ends with $ without any additional slashes before and afterwards',
+          'excluded_patterns must be an array of regex that starts with ^/ and ends with $ for example ^/blog/[d]{4}$',
```
+1 to @eduardoboucas's suggestion here too. Though I do want to re-ask my open questions from https://github.com/netlify/edge-bundler/pull/339#discussion_r1140583208 because there are some discrepancies between this error message and [a related docs update](https://github.com/netlify/docs/pull/2943) and I'd like to figure out how to resolve that

1. Is `excluded_patterns` an internal syntax or is this the property that users and frameworks should be using? I ask because the [excluded patterns demo](https://github.com/netlify/excluded-patterns-demo) that I referenced while [documenting exclusions](https://github.com/netlify/docs/pull/2943) uses the syntax `excludedPattern` (camel case singular vs snake case plural)

2. Are customers/frameworks able to enter an array of regexes for excludedPattern or is the support for arrays here an internal implementation detail (maybe for handling multiple declarations for the same function)? If customers/frameworks can enter an array of regexes what is the syntax for doing that in `netlify.toml` and `manifest.json`? I ask because the [excluded patterns demo](https://github.com/netlify/excluded-patterns-demo) that I referenced while [documenting exclusions](https://github.com/netlify/docs/pull/2943) didn't indicate support for arrays. Here's a table of my understanding of what is supported for a single declaration in the various formats (this understanding is what my open docs PR is based on https://github.com/netlify/docs/pull/2943). Please correct anything that I've got wrong

|   | inline Config  | netlify.toml  | manifest.json  |
|---|---|---|---|
| path  | string or array of strings  | single string  |  single string |
| excludedPath |  string or array of strings |  single string |  single string |
| pattern  | not supported  | single regex  |  single regex |
| excludedPattern  | not supported | single regex  | single regex  |

danez replied:
@eduardo should `excludePath` be allowed in `netlify.toml` or is there a reason it is not there?

klavavej commented on node/validation/manifest/schema.ts:
```diff
@@ -35,7 +34,7 @@ const functionConfigSchema = {
         format: 'regexPattern',
         errorMessage:
-          'excluded_patterns needs to be an array of regex that starts with ^ followed by / and ends with $ without any additional slashes before and afterwards',
+          'excluded_patterns must be an array of regex that starts with ^/ and ends with $ (e.g. ^/blog/[d]{4}$)',
```
If this is meant to be a customer-facing error message, I think it would be helpful to use the customer-facing input name rather than the name of the internal property we validate. Using the customer-facing name should help folks find and fix the invalid value that's triggering the error message.

```suggestion
          'excludedPattern must be an array of regex that starts with ^/ and ends with $ (e.g. ^/blog/[d]{4}$)',
```

danez replied:
Hmm.. The worst part about this validation is that it will most likely not be fixable by the user. 

1. If a user misses the slash in `netlify.toml` the config validation at the beginning of the build will fail with an error message.
```
11:32:34 AM: Failed during stage 'Reading and parsing configuration files': 
When resolving config file /opt/build/repo/netlify.toml:
Configuration property edge_functions[1].path must be a valid path.

Invalid syntax

  [[edge_functions]]
  path = "hello/*"

Valid syntax

  [[edge_functions]]
  path = "/hello"
  function = "hello"

: exit status 1
```

2. If the slash is missing in the in-source-config then TypeScript will warn about that.

So it is pretty unlikely that this error comes from invalid user input, but more likely from an integration that generates a pattern without a slash. So I think a user will in 90% not know what to do, no matter how we name it. 

I wonder if we should instead of validating the manifest should have validation for edge function declarations from integrations, so that we can point the user to `an integration was declaring an invalid edge function`

klavavej replied:
> I wonder if we should in addition of validating the manifest have validation for edge function declarations from integrations, so that we can point the user to `an integration was declaring an invalid edge function`

I like this idea! This could give the user a more actionable step forward (remove the integration or contact the integration provider with a bug report). It would be great if an error message like this could provide `generator` info for the problematic function to help users figure out which integration to remove / contact with a bug report.

danez commented:
We decided to not validate the leading slash in `pattern` and `excludePattern`, as there might be cases where there are valid reasons not to start with slash like `^(?:/path1|/path2)$` and there is also not real gain for us nor for the integrator to enforce this check.


This means that in this PR I reverted the FF for the check and removed the actual check again. The new message is still valid, just without the slash.

klavavej reviewed (commented):
revised error message looks good to me
